nameko/messaging.py

```diff
@@ -149,15 +148,16 @@ def retry_policy(self):
         return DEFAULT_RETRY_POLICY
 
     def setup(self):
-
         exchange = self.exchange
-        queue = self.queue
+        queues_to_declare = self.queues_to_declare
 
         verify_amqp_uri(self.amqp_uri)
 
         with get_connection(self.amqp_uri) as conn:
-            if queue is not None:
-                maybe_declare(queue, conn)
+            if queues_to_declare is not None:
+                for queue in queues_to_declare:
+                    if queue is not None:
+                        maybe_declare(queue, conn)
```

Comment:
Thanks for the suggestion. As far as I see it, the big change here would be that we will ensure queues and exchanges are correctly bound before publishing a message rather than just once when the dependency is set up, so Kombu will have to declare those entities each time.

I am happy to change it and let Kombu do it for us when publishing, if you think those extra declarations are acceptable.